airflow/providers/hashicorp/hooks/vault.py

```diff
@@ -0,0 +1,308 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+"""Hook for HashiCorp Vault"""
+import json
+from typing import Optional, Tuple
+
+import hvac
+from hvac.exceptions import VaultError
+from requests import Response
+
+from airflow.hooks.base_hook import BaseHook
+from airflow.providers.hashicorp.common.vault_client import VaultClient
+
+DEFAULT_KUBERNETES_JWT_PATH = '/var/run/secrets/kubernetes.io/serviceaccount/token'
+DEFAULT_KV_ENGINE_VERSION = 2
+
+
+class VaultHook(BaseHook):
```

Comment:
Not sure but maybe `HashiCorpVaultHook` will be more meaningful?

Reply:
We alreday have hashicorp in the package name. I think it is enough. Vault has become a household name - for example it's the HashiCorp Vault which is marketed as Azure Vault (but it's nothing more that HashiCorp Vault that' is centrally managed by Microsoft on Azure). 

I think VaultHook is good in this case :)